cmd/clusterctl/cmd/config_cluster.go

```diff
@@ -0,0 +1,96 @@
+	Use:   "cluster",
+	Short: "Generate templates for creating a Cluster API workload clusters",
+	Long: LongDesc(`
+		clusterctl ships with a list of well know providers; if necessary, edit
+		the $HOME/.clusterctl.yaml file to add new provider configurations or to customize existing ones.
+		
+		Each provider configuration links to a repository, and clusterctl will fetch the template
+		from creating a new cluster from there.`),
+
+	Example: Examples(`
+		# Generates a yaml file for creating a Cluster API workload clusters using
+		# default infrastructure provider and default bootstrap provider installed in the cluster.
+		clusterctl config cluster my-cluster
+ 
+		# Generates a yaml file for creating a Cluster API workload clusters using
+		# specified infrastructure and bootstrap provider
+		clusterctl config cluster my-cluster --infrastructure=aws --bootstrap=kubeadm
+ 
+		# Generates a yaml file for creating a Cluster API workload clusters using
+		# specified version of the AWS infrastructure provider
+		clusterctl config cluster my-cluster --infrastructure=aws:v0.4.1
+
+		# Generates a yaml file for creating a Cluster API workload clusters in the "foo" namespace.
+		clusterctl config cluster my-cluster --target-namespace=foo
+		
+		# Generates a yaml file for creating a Cluster API workload clusters with
+		# nodes using Kubernetes v1.16.0
+		clusterctl config cluster my-cluster --kubernetes-version=v1.16.0
+
+		# Generates a yaml file for creating a Cluster API workload clusters with
+		# custom number of nodes (if supported by provider's templates)
+		clusterctl config cluster my-cluster --controlplane-machine-count=3 --worker-machine-count=10`),
+
+	Args: cobra.ExactArgs(1),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return runGenerateCluster(args[0])
+	},
+}
+
+func init() {
+	configClusterClusterCmd.Flags().StringVarP(&cc.kubeconfig, "kubeconfig", "", "", "Path to the kubeconfig file to use for accessing the management cluster. If empty, default rules for kubeconfig discovery will be used")
+
+	configClusterClusterCmd.Flags().StringVarP(&cc.infrastructureProvider, "infrastructure", "i", "", "The infrastructure provider that should be used for creating the workload cluster")
+	configClusterClusterCmd.Flags().StringVarP(&cc.bootstrapProvider, "bootstrap", "b", "kubeadm", "The provider that should be used for bootstrapping K8s nodes in the workload cluster")
+
+	configClusterClusterCmd.Flags().StringVarP(&cc.flavor, "flavor", "f", "", "The template variant to be used for crating the workload cluster")
+	configClusterClusterCmd.Flags().StringVarP(&cc.targetNamespace, "target-namespace", "", "", "The namespace where the objects describing the workload cluster should be deployed. If not specified, the current namespace will be used")
+	configClusterClusterCmd.Flags().StringVarP(&cc.kubernetesVersion, "kubernetes-version", "", "1.15.0", "The Kubernetes version to use for the workload cluster")
```

Comment:
Makes sense. I'm going to remove the default.
In case the flag will be set, I will use it, otherwise, I will read from the config file/env variables, otherwise fail